canvas_modules/common-canvas/src/common-canvas/svg-canvas-utils-decs.js

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2021 Elyra Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { has } from "lodash";
+import { DEC_LINK, TEXT_AREA_BORDER_ADJUSTMENT } from "./constants/canvas-constants";
+
+export default class SvgCanvasDecs {
+	constructor(canvasLayout) {
+		this.canvasLayout = canvasLayout;
+	}
+
+	getDec(id, decorations) {
+		if (decorations) {
+			const dec = decorations.find((nd) => nd.id === id);
+			return dec;
+		}
+		return null;
+	}
+
+	getDecLabelForeignClass(dec) {
+		const outlineClass = dec.label_outline ? " d3-node-label-outline" : "";
```

Comment:
Should this have a different class name from node? This is adding a border for this label, should it be bigger so the edit icon can fit inside or should there be no borders for link labels?

<img width="144" alt="image" src="https://user-images.githubusercontent.com/1628100/121086205-b7a58800-c797-11eb-9984-959b3e1d525f.png">

Reply:
Hi @caritaou Good catch, however the label outline is just for debug purposes which is why I didn't document the `label_outline` property in the wiki. It's just so I can see the extent of the div. If the host app wants an outline around the label they would need to specify a `shape` decoration in addition to the label decoration.